pkg/manager/manager_test.go

```diff
@@ -1675,6 +1677,112 @@ var _ = Describe("manger.Manager", func() {
 		Expect(err).NotTo(HaveOccurred())
 		Expect(m.GetAPIReader()).NotTo(BeNil())
 	})
+
+	Context("with logical clusters", func() {
+		It("should be able to create a manager with a logical cluster", func() {
+			adapter := &fakeLogicalAdapter{
+				list: []logical.Name{
+					"test-cluster",
+				},
+			}
+
+			m, err := New(cfg, Options{}.WithExperimentalLogicalAdapter(adapter))
+			Expect(err).NotTo(HaveOccurred())
+
+			var built atomic.Bool
+			var started atomic.Bool
+			err = m.AddLogicalRunnableBuilder(func(name logical.Name, cl cluster.Cluster) (Runnable, error) {
+				built.Store(true)
+				return RunnableFunc(func(ctx context.Context) error {
+					Expect(logical.FromContext(ctx)).To(Equal(logical.Name("test-cluster")))
+					started.Store(true)
+					<-ctx.Done()
+					return nil
+				}), nil
+			})
+			Expect(err).NotTo(HaveOccurred())
+
+			By("starting the manager")
+			ctx, cancel := context.WithCancel(context.Background())
+			doneCh := make(chan struct{})
+			go func() {
+				defer GinkgoRecover()
+				defer close(doneCh)
+				Expect(m.Start(ctx)).To(Succeed())
+			}()
+			<-m.Elected()
+
+			Eventually(built.Load).Should(BeTrue())
+			Eventually(started.Load).Should(BeTrue())
+			By("making sure there's no extra go routines still running after we stop")
+			cancel()
+			<-doneCh
+		})
+
+		It("should cancel only the removed cluster runnables", func() {
+			adapter := &fakeLogicalAdapter{
+				watch: make(chan logical.Event),
+				list: []logical.Name{
+					"test-cluster",
+					"removed-cluster",
+				},
+			}
+
+			m, err := New(cfg, Options{}.WithExperimentalLogicalAdapter(adapter))
+			Expect(err).NotTo(HaveOccurred())
+
+			var built atomic.Int64
+			var started atomic.Int64
+			var completed atomic.Int64
+			removedCh := make(chan struct{})
+			err = m.AddLogicalRunnableBuilder(func(name logical.Name, cl cluster.Cluster) (Runnable, error) {
+				built.Add(1)
+				return RunnableFunc(func(ctx context.Context) error {
+					defer completed.Add(1)
+					started.Add(1)
+					switch logical.FromContext(ctx) {
+					case "test-cluster":
+						// All good!
+					case "removed-cluster":
+						// This should be canceled.
```

Comment:
nit: not sure if this test would notice if we're also hitting the test-cluster case